lib/rubocop/cop/rails/save_bang.rb

```diff
@@ -221,15 +221,20 @@ def check_used_in_conditional(node)
           true
         end
 
-        def conditional?(node) # rubocop:disable Metrics/CyclomaticComplexity
+        def in_boolean_expr?(node)
           node = node.block_node || node
+          parent = node.parent
+          return false unless parent
 
-          condition = node.parent
-          return false unless condition
+          predicate_node?(parent) || condition_of?(parent, node)
+        end
+
+        def predicate_node?(node)
+          node.or_type? || node.and_type? || single_negative?(node)
+        end
 
-          condition.if_type? || condition.case_type? ||
-            condition.or_type? || condition.and_type? ||
-            single_negative?(condition)
+        def condition_of?(parent, node)
```

Comment:
It may be a little difficult to infer the behavior from the method name. Can you improve it?